app/workers/github_deployment_watchdog_worker.rb

```diff
@@ -0,0 +1,45 @@
+class GithubDeploymentWatchdogWorker
+  include Sidekiq::Worker
+  sidekiq_options queue: 'deployment_watchdog'
+
+  # default time to wait until our worker wakes up.
+  DEFAULT_DELAY = 30.seconds
+
+  # create a class method to hardcode how we want to schedule this worker.
+  # user_id: the database id of the slashdeploy User object.
+  # github_repo: the repo of the Github Deployment.
+  # github_deployment_id: the id of the external Github Deployment.
+  def self.schedule(user_id, github_repo, github_deployment_id)
+    self.perform_in(DEFAULT_DELAY, user_id, github_repo, github_deployment_id)
+  end
+
+  # notify the user if there was an issue otherwise do nothing.
+  # user_id: the database id of the slashdeploy User object.
+  # github_repo: the repo of the Github Deployment.
+  # github_deployment_id: the id of the external Github Deployment.
+  def perform(user_id, github_repo, github_deployment_id)
+    # get the slashdeploy User object.
+    user = User.find(user_id)
+
+    # get the Github Deployment by id.
+    github_deployment = SlashDeploy.service.github.get_deployment(user, github_repo, github_deployment_id)
```

Comment:
`SlashDeploy::Service#github` is assumed private (we should actually explicitly flag it as private). Let's wrap these two lines up in a method on `SlashDeploy::Service`:

```ruby
# Returns truthy if the given deployment has started (has deployment statuses)
def deployment_started?(user, repo, deployment_id)
  deployment = github.get_deployment(user, repo, deployment_id)
  deployment_status = github.last_deployment_status(user, github_deployment.url)
  deployment if deployment_status
end
```